weather-wise/src/main/java/app/weatherwise/APIService.java

```diff
@@ -0,0 +1,72 @@
+package app.weatherwise;
+
+// import toolkits to decode JSON
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+// import toolkits to establish HTTP connections and retrieve data from URLs
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+// other imports
+import java.util.Scanner;
+import java.io.IOException;
+
+public class APIService {
+
+    private final String apiUrl;
+
+    public APIService(String url) {
+        this.apiUrl = url;
+    }
+
+    // creates an HTTP request to specified query and returns response as a String
+    private String makeHttpRequest() throws IOException {
+        URL url = new URL(apiUrl);
+        HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection(); // open a connection to the url using HTTP
+        urlConnection.setRequestMethod("GET"); // set request method to GET (to retrieve data)
+        urlConnection.connect(); // establish connection to url
+
+        int responseCode = urlConnection.getResponseCode();
+        String responseMessage = urlConnection.getResponseMessage();
+        if (responseCode != 200) { // if responseCode != 200 (200 is a success response code) then throw exception
+            throw new RuntimeException("Connection failed with HTTP response code: " + responseCode + "\n" + responseMessage);
+        } else {
+            StringBuilder weatherInfo = new StringBuilder();
+            Scanner scanner = new Scanner(url.openStream());
+            while (scanner.hasNext()) { // read the JSON data from the API response
+                weatherInfo.append(scanner.nextLine()); // append to weatherInfo
+            }
+            scanner.close();
+
+            return weatherInfo.toString();
+        }
+    }
+
+    // parse the response and returns as an equivalent JSONObject object
+    public JSONObject getJSONObject() {
+        try {
+            String weatherInfo = makeHttpRequest();
+            JSONParser parse = new JSONParser();
+            return (JSONObject) parse.parse(weatherInfo);
+        } catch (Exception e) {
+            System.err.println("Unknown error occurred: " + e.getMessage());
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    // parse the response and returns as an equivalent JSONArray object
+    public JSONArray getJSONArray(){
+        try {
+            String informationString = makeHttpRequest();
+            JSONParser parse = new JSONParser();
+            return (JSONArray) parse.parse(informationString);
+        } catch (Exception e) {
+            System.err.println("Unknown error occurred: " + e.getMessage());
+            e.printStackTrace();
+        }
+        return null;
+    }
+}
```

Comment:
Big addition to Weather Wise, the **APIService** class. This is now the main class that will send HTTP requests to the OpenWeatherMap API and return the response using Java's networking classes:
`import java.net.HttpURLConnection;`
`import java.net.URL;`

It also uses the previously mentioned JSON Simple library classes to create JSONObject objects and JSONArray objects which is used to parse the response JSON String of the API into mutable java objects:
`import org.json.simple.JSONArray;`
`import org.json.simple.JSONObject;`
`import org.json.simple.parser.JSONParser;`

Since this class is not hard-coded to only work with the OpenWeatherMap API, it can be used in other projects as long as the correct dependency is attached.

Reply:
To add to the previous comment, there is also simple exception catching in the case of invalid arguments provided to the API. Invalid parameters that are specific to the weather API will eventually be handled in the Controller itself.